File: HashTable/219_Contains_Duplicate_II/219_Contains_Duplicate_II.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> d = new HashMap<>();
        for(int i=0; i<nums.length; i++) {
            if(d.containsKey(nums[i]) && Math.abs(i-d.get(nums[i])) <=k)
                return true;
            d.put(nums[i], i);
        }

        return false;
    }
}

File: HashTable/219_Contains_Duplicate_II/219_Contains_Duplicate_II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> d;
        for(int i=0; i<nums.size(); i++) {
            if (d.find(nums[i]) != d.end() && abs(i-d[nums[i]])<=k)
                return true; 
            d[nums[i]] = i; 
        }

        return false;
    }
};